realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -204,6 +204,16 @@ public void removeColumn(long columnIndex) {
     @Override
     public void renameColumn(long columnIndex, String newName) {
         verifyColumnName(newName);
+        if (getPrimaryKey() == columnIndex) {
+            Table pkTable = getPrimaryKeyTable();
+            String className = tableNameToClassName(getName());
+            long rowIndex = pkTable.findFirstString(PRIMARY_KEY_CLASS_COLUMN_INDEX, className);
+            if (rowIndex != NO_MATCH) {
+                pkTable.getUncheckedRow(rowIndex).setString(PRIMARY_KEY_FIELD_COLUMN_INDEX, newName);
```

Comment:
You can just call `setString` directly on the table. No need to create an unchecked row for it.